packages/page-settings/src/modals/NetworkSpecs.tsx

```diff
@@ -0,0 +1,194 @@
+  const _onChangeValue = (k: keyof NetworkSpecsStruct): inputListener => (v: string): void => setNetworkSpecs({ [k]: v });
+  const _onSetRandomColor = (): void => setNetworkSpecs({ color: getRandomColor() });
+  const _checkPathIdValid = (): boolean => /^[\w-.]+$/.test(networkSpecs.pathId);
+  const _checkColorValid = (): boolean => /^#[\da-fA-F]{6}|#[\da-fA-F]{3}$/.test(networkSpecs.color);
+
+  return <Modal
+    className={className}
+    header={t('Export Network Specs')}
+  >
+    <Modal.Content>
+      <Input
+        className='full'
+        help={t('Name of the network. It only for display purpose.')}
+        label={t('Network Name')}
+        onChange={_onChangeValue('title')}
+        value={networkSpecs.title}
+      />
+      <Input
+        autoFocus
+        className='full'
+        help={t('the path id used as the path prefix when deriving new account, all the accounts under this network will have the same prefix')}
+        isError={!_checkPathIdValid()}
+        label={t('Path ID')}
+        onChange={_onChangeValue('pathId')}
+        value={networkSpecs.pathId}
+      />
+      <Input
+        className='full'
+        help={t('The color used to distinguish this network with others, use color code with 3 or 6 digits, like "#FFF" or "#111111"')}
+        isError={!_checkColorValid()}
+        label={t('Color')}
+        onChange={_onChangeValue('color')}
+        value={networkSpecs.color}
+      >
+        <div className='settings--networkSpecs-colorButton'>
+          <Button
+            label={t('Generate Random Color')}
+            icon='sync'
+            key='spread'
+            onClick={_onSetRandomColor}
+          />
+          <ColoredComponent color={networkSpecs.color}/>
+        </div>
+      </Input>
+      <Input
+        className='full'
+        help={t('Genesis Hash refers to initial state of the chain, it cannot be changed once the chain is launched')}
+        isDisabled
+        label={t('Genesis Hash')}
+        value={networkSpecs.genesisHash}
+      />
+      <Input
+        className='full'
+        help={t('Unit decides the name of 1 unit token, e.g. "DOT" for Polkadot')}
+        isDisabled
+        label={t('Unit')}
+        value={networkSpecs.unit}
+      />
+      <Input
+        className='full'
+        help={t('Prefix indicates the ss58 address format in this network, is a number between 0 ~ 255 describes the precise format of the bytes of the address')}
+        isDisabled
+        label={t('Address Prefix')}
+        value={networkSpecs.prefix}
+      />
+      <Input
+        className='full'
+        help={t('Decimals decides the smallest unit of the token, which is 1/10^decimals')}
+        isDisabled
+        label={t('Decimals')}
+        value={networkSpecs.decimals}
+      />
+      <QrNetworkSpecs className='settings--networkSpecs-qr' networkSpecs={debouncedQrData}/>
+    </Modal.Content>
+    <Modal.Actions onCancel={onClose} withOr={false}>
```

Comment:
One thing I am not sure is that how do I use `<Modal.Actions>` with only a "cancel" button, lint gives warning that there is no children

Reply:
Was not really to be meant to be used without. For now just add an explicit close as well (to work around this), since we will combine this into the "external sources" thing and drop the modal.